src/accel/FastSimulationOffload.cc

```diff
@@ -0,0 +1,88 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/FastSimulationOffload.cc
+//---------------------------------------------------------------------------//
+#include "FastSimulationOffload.hh"
+
+#include "corecel/Assert.hh"
+
+#include "ExceptionConverter.hh"
+#include "LocalTransporter.hh"
+#include "SharedParams.hh"
+
+namespace celeritas
+{
+FastSimulationOffload::FastSimulationOffload(G4String const& name,
+                                             SharedParams const* params,
+                                             LocalTransporter* local)
+    : G4VFastSimulationModel(name), params_(params), transport_(local)
+{
+    CELER_VALIDATE(G4VERSION_NUMBER > 1110,
+                   << "the current version of Geant4 (" << G4VERSION_NUMBER
+                   << ") is too old to support the fast simulation Flush() "
+                      "interface");
+    CELER_EXPECT(params);
+    CELER_EXPECT(local);
```

Comment:
To reduce duplication, you could have this forward (assuming the base class takes a nullptr if the region is unspecified?)
```suggestion
    : FastSimulationOffload(name, nullptr, params, local)
{
```

Reply:
This isn't possible as the `G4VFastSimulationModel::G4VFastSimulationModel(const G4String&, G4Envelope*, G4bool)` constructor requires the `G4Envelope` pointer to be not `nullptr`. It doesn't check this though, and yes, these constructors are pretty old and in need of modernisation!